### ISSUE TITLE
Changelog is outdated

### ISSUE THREAD

**Issue by ChristianIvicevic** [shikijs/shiki#223]
In the past months the library is getting updates but they is no user-friendly documentation since the changelog in the repository is outdated. It'd be appreciated if you take the time to keep it up-to-date so that one does not need to read through the commit history to deduce what has changed and whether there are breaking changes.

**Comment by orta** [comment]
I don't have the time to make a nuanced changelog for shiki, happy to accept some automation improvements here 👍🏻

**Comment by octref** [comment]
Let's do Conventional Commits and auto-generate changelogs: #230.

**Comment by orta** [comment]
Personally I'm not a fan of forcing everyone to adopt a strange commit formatting system, especially when the actual abstraction unit we're interested in documenting is at a per-PR level. https://artsy.github.io/blog/2019/01/03/label-based-prs/

**Comment by octref** [comment]
@orta Hey long time no see! Just moved to Boston from Shanghai so I disappeared for a while. Sorry about that.

I chose CC because I didn't know any other alternatives. I only care about having a convention to make the messages machine-readable so a tool can generate a changelog. PR Labels seems like more work on infrastructure.

Essentially I just want each commit pre-fixed with fix/feat/docs etc, and a tool to easily generate a changelog from it.

**Comment by Gerrit0** [comment]
I agree with orta here - I did conventional commits with TypeDoc for a few months, complete with architecture to [generate the changelog](https://github.com/TypeStrong/typedoc/blob/v0.21.9/scripts/generate_changelog.js) and [automatically create](https://github.com/TypeStrong/typedoc/blob/v0.21.9/scripts/create_release.js) a GitHub release containing it. I recently dropped this in favor of maintaining a changelog.md file in the package root, and updating that as work is done. (Tooling automatically pulls out the appropriate section to publish as the GitHub release still)

This has been much less of a pain to work with since the changelog can go into more detail than the first line of a commit message, can reorder features/bug fixes so that more prominent ones are included at the top, and can be updated after the fact to address anything that was missed in initial commits. Then, adding a checkbox to the pull request template for updated changelog is easy.
